### PR TITLE
http connection manager proto: raise max_request_headers_kb limit to 8192 KiB (8MiB) from 96 KiB

### PR DESCRIPTION
**Pull request by anirudhaps**
Signed-off-by: Anirudha Singh <ps.anirudha@gmail.com>

Commit Message:
Raise max configurable `max_request_headers_kb` limit to 8192 KiB (8MiB) from 96 KiB in http connection manager protobuf. 

Additional Description:
Added/Updated relevant unit tests and integration tests. This change will allow increasing `max_request_headers_kb` to 8MiB from http connection manager's configuration. The change is mainly updating the limit in a validation check in the protobuf. Also, the old (merged) PR [#5859](https://github.com/envoyproxy/envoy/pull/5859) is read, no nghttp2 library-related issues are observed on raising the `max_request_headers_kb` beyond 96 KiB.
Risk Level: Low
Testing: Unit, Integration, Manual
Docs Changes: Inline in proto file.

### CODE REVIEW

**Comment by alyssawilk** [comment]
cc @auni53 can you remind me what the implementation constraints were?  I thought one of the parsers didn't support larger headers so maybe it has been addressed?

**Comment by antoniovicente** [comment]
> cc @auni53 can you remind me what the implementation constraints were? I thought one of the parsers didn't support larger headers so maybe it has been addressed?

I thought there was a limitation in the H1 parser but e2e tests do suggest that this works e2e for H1 and H2.

File: test/integration/protocol_integration_test.cc [hunk 1]
```diff
@@ -1578,11 +1578,23 @@ TEST_P(DownstreamProtocolIntegrationTest, LargeRequestHeadersRejected) {
   testLargeRequestHeaders(95, 1, 60, 100);
 }
 
+TEST_P(DownstreamProtocolIntegrationTest, VeryLargeRequestHeadersRejected) {
```
Comment (alyssawilk) [inline test/integration/protocol_integration_test.cc:1581]: I'd think you'd need to exclude this for H3 as well, as we don't have header configuration limits.  We'll see what CI says :-)

File: api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto [hunk 2]
```diff
@@ -299,10 +299,8 @@ message HttpConnectionManager {
   // The maximum request headers size for incoming connections.
   // If unconfigured, the default max request headers allowed is 60 KiB.
   // Requests that exceed this limit will receive a 431 response.
-  // The max configurable limit is 96 KiB, based on current implementation
-  // constraints.
   google.protobuf.UInt32Value max_request_headers_kb = 29
```
Comment (alyssawilk) [inline api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto:302]: mind adding a release note (docs/root/version_history/current.rst) about the change?

**Review by alyssawilk — COMMENTED** [review]
Looks good - just one (maybe two pending CI) nits :-)

**Comment by htuch** [comment]
Will this work with llhttp for H1 parser? CC @asraa

File: test/integration/protocol_integration_test.cc (hunk 1 above)
Comment (anirudhaps) [reply test/integration/protocol_integration_test.cc:1581]: Added exclude macros

File: api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto (hunk 2 above)
Comment (anirudhaps) [reply api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto:302]: Added release notes

**Review by alyssawilk — APPROVED** [review]
LGTM pending happy CI, but I'm going to add @asraa per Harvey's point about the other library

**Comment by asraa** [comment]
> Will this work with llhttp for H1 parser? CC @asraa

Yes! Should be fine. We handle it in the codec code, and the parser knows that https://github.com/nodejs/llhttp/issues/2

**Comment by anirudhaps** [comment]
Updated docs/root/version_history/current.rst as per latest main to avoid any cause of merge conflict

**Review by alyssawilk — COMMENTED** [review]
@htuch looks like this needs another API stamp, though I assume the v2 changes be reverted for that?

**Comment by auni53** [comment]
Sorry to comment late, yes when I was trying to raise the limits the barrier was in the HTTP1 codec library that we use.

**Review by htuch — COMMENTED** [review]
/lgtm api

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**